model/project_parser.go

```diff
@@ -644,40 +650,74 @@ func LoadProjectInto(ctx context.Context, data []byte, opts *GetProjectOpts, ide
 	}
 
-	// return intermediateProject even if we run into issues to show merge progress
-	for _, path := range intermediateProject.Include {
+	if len(intermediateProject.Include) > 0 {
 		if opts == nil {
 			err = errors.New("trying to open include files with empty options")
 			return nil, errors.Wrapf(err, LoadProjectError)
 		}
-		opts.UpdateForFile(path.FileName)
-
-		var yaml []byte
-		opts.Identifier = identifier
-		opts.RemotePath = path.FileName
-		grip.Debug(message.Fields{
-			"message":     "retrieving included YAML file",
-			"remote_path": opts.RemotePath,
-			"read_from":   opts.ReadFileFrom,
-			"module":      path.Module,
-		})
-		if path.Module != "" {
-			yaml, err = retrieveFileForModule(ctx, *opts, intermediateProject.Modules, path.Module)
-		} else {
-			yaml, err = retrieveFile(ctx, *opts)
-		}
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, path.FileName)
-		}
-		add, err := createIntermediateProject(yaml, opts.UnmarshalStrict)
-		if err != nil {
-			return intermediateProject, errors.Wrapf(err, "%s: loading file '%s'", LoadProjectError, path.FileName)
+		wg := sync.WaitGroup{}
+		catcher := grip.NewBasicCatcher()
+		yamlNamePairs := make(chan yamlNamePair, len(intermediateProject.Include))
+
+		for _, path := range intermediateProject.Include {
+			wg.Add(1)
+			go func(include Include, projectOpts *GetProjectOpts) {
+				defer wg.Done()
+				// Make a copy of opts because otherwise parts opts would be
+				// modified concurrenly.  Note, however, that Ref and PatchOpts are
+				// themselves pointers, so should not be modified.
+				localOpts := &GetProjectOpts{
+					Ref:             opts.Ref,
+					PatchOpts:       opts.PatchOpts,
+					LocalModules:    opts.LocalModules,
+					RemotePath:      projectOpts.RemotePath,
+					Revision:        projectOpts.Revision,
+					Token:           projectOpts.Token,
+					ReadFileFrom:    projectOpts.ReadFileFrom,
+					Identifier:      projectOpts.Identifier,
+					UnmarshalStrict: projectOpts.UnmarshalStrict,
+				}
+				localOpts.UpdateForFile(include.FileName)
+
+				var yaml []byte
+				localOpts.Identifier = identifier
+				localOpts.RemotePath = include.FileName
+				grip.Debug(message.Fields{
+					"message":     "retrieving included YAML file",
+					"remote_path": localOpts.RemotePath,
+					"read_from":   localOpts.ReadFileFrom,
+					"module":      include.Module,
+				})
+				if include.Module != "" {
+					yaml, err = retrieveFileForModule(ctx, *localOpts, intermediateProject.Modules, include.Module)
+					if err == nil && yaml != nil {
+						yamlNamePairs <- yamlNamePair{yaml, include.FileName}
+					}
+				} else {
+					yaml, err = retrieveFile(ctx, *localOpts)
+					if err == nil && yaml != nil {
+						yamlNamePairs <- yamlNamePair{yaml, include.FileName}
+					}
+				}
+				catcher.Add(errors.Wrapf(err, "%s: retrieving file '%s'", LoadProjectError, include.FileName))
```

Comment:
Rather than doing this in an if-else, where there are further if blocks within the if else, can we just make this more streamlined so there's one case and another case? The only shared line is the lingering `catcher` adding at the end, so it doesn't seem worth it to complicate the logic:
```go
if include.Module != "" {
    yaml, err := ...
    if yaml != nil {
        yamlNamePairs <- ...
    }
    catcher.Wrapf(...)
    return
}

yaml, err := retrieveFile(...)
if yaml != nil {
    yamlNamePairs <- ...
}
catcher.Wrapf(...)
```

Reply:
This went away with the helper function.